homeassistant/components/utility_meter/__init__.py

```diff
@@ -92,13 +93,22 @@ async def async_setup(hass, config):
             # add one meter for each tariff
             tariff_confs = []
             for tariff in conf[CONF_TARIFFS]:
-                tariff_confs.append(
-                    {
-                        CONF_METER: meter,
-                        CONF_NAME: f"{meter} {tariff}",
-                        CONF_TARIFF: tariff,
-                    }
-                )
+                if name:
```

Comment:
There is no need for this `if`

The `get` in line 69 will ensure that name will get `meter` as default value

Reply:
I understand on the breaking change, it does change the name that shows in the UI, but does not change the entity_id so it seems to be a small effect, not causing automation breakage, etc.

I didn't understand the behavior of line 69 getting that meter as default, I'll rework to simplify and pull out the if.

Thanks!

Reply:
It can change if you use something in `name` that does not match the current slug

Reply:
That's interesting, I'm not sure I understand. If you have a config like this:

```
utility_meter:
  energy_used:
    name: "Foobar"
    source: sensor.energy_daily
    cycle: daily
    tariffs:
      - peak
      - offpeak
```
Are you saying that the sensors created will have entity_id's that are formed by the `name:` (i.e. `sensor.foobar_peak` and `sensor.foobar_offpeak` in this case) and not the `energy_used:` part? That could definitely be a problem if it is true.

Reply:
I just verified it breaks when the name doesn't match the slug, I'm not sure how to proceed, please let me know what you think should be done.

Thanks!

Reply:
Please write a paragraph in the PR description indicating this Breaking change so that it can go into the release notes.

Reply:
I will create a good breaking change description if that's how we want to go. I think most people will be OK with it but it could trip someone up if they don't pay attention.

Reply:
Please have a look at the breaking change explanation I've added and see if it makes sense.

Thanks!

Reply:
Very good description :)

But... for release notes we need a TLDR; version ;)

Reply:
My proposal:

"Tariff sensors will inherit a their name from the utility meter name and not from the the utility meter slug. Your sensor entity names might change cause of this, keep a utility_meter name based on the slug to avoid this."

Reply:
Sorry, I do get verbose sometimes.  :)
I like your version, thanks for the help!

Reply:
I will try to figure out how to decouple the slug and entity_id from the name to make it a bit less confusing for a future change.

Reply:
I don't think it's worth it, if people want to keep previous sensors they just need to follow the breaking change advise (please update the description)

Reply:
Ok, makes sense, I've changed the title, if it is not clear please let me know. Thanks!